style_editor.html

```diff
@@ -10018,16 +10077,117 @@
+
+            <label title="Simulates In-Hilt LED blades, (no addressable pixel effects).">
+              <input type="checkbox" id="INHILT_BUTTON" name="inhilt"
+		data-state="inhiltState" onclick="handleClick(this)">
+              <span>Inhilt LED</span>
+            </label>
+
+            <label title="Provides &quot;snapshots&quot; of rapidly moving things to see what they look like without fast movement. *Note* Graflex hilt model probably won't be able to render with this on.">
+              <input type="checkbox" id="SLOW_BUTTON" name="slow"
+		data-state="slowState" onclick="handleClick(this)">
+              <span>Slow</span>
+            </label>
+          </div>
+        </div>
+<script>
+
+function toggleSettingsPanel() {
+  var settingsButton = FIND("SETTINGS_BUTTON");
+  var settingsPanel = FIND("settings_panel");
+
+  settingsPanel.classList.toggle("show");
+
+  // Mouseleave event listener
+  if (settingsPanel.classList.contains("show")) {
+    var timeoutId = null;
+    settingsPanel.addEventListener("mouseleave", function(e) {
+      timeoutId = setTimeout(function() {
+        settingsPanel.classList.remove("show");
+      }, 1000);
+    });
+    settingsPanel.addEventListener("mouseenter", function(e) {
+      if (timeoutId) {
+        clearTimeout(timeoutId);
+        timeoutId = null;
+      }
+    });
+  }
+}
+
+// Call the onPageLoad function when the page is loaded
+window.addEventListener('DOMContentLoaded', onPageLoad);
+
+function onPageLoad() {
+  inhiltState.onload();
+  slowState.onload();
+}
+
+/* Settings buttons saved as local storage */
+function getSavedState(buttonState, defaultValue) {
+  var value = localStorage.getItem(buttonState);
+  console.log("Retrieved SavedState for " + buttonState + ": " + value);
+
+  if (value === null) {
+    return defaultValue;
+  }
+  // Convert the string to boolean unless it's null or "false"
+  return value !== "false";
+}
+
+function saveState(buttonState, settingIsOn) {
+  localStorage.setItem(buttonState, settingIsOn);
+}
+  
+var all_saved_states = [];
+
+class SavedState {
+  constructor(name, def, update_function) {
+    this.name = name;
+    this.def = def;
+    this.update_function = update_function;
+    all_saved_states.push(this);
+  }
+  onload() {
+    this.set(getSavedState(this.name+"Save", this.def));
+  }
+  set(value) {
+    this.value = value;
+    FIND(this.name.toUpperCase()+"_BUTTON").checked = value ? true : false;
+    saveState(this.name+"Save", value);
+    this.update_function(value);
+  }
+  get() { return this.value; }
+}
+
+var inhiltState = new SavedState("inhilt", false, (on) => { STATE_NUM_LEDS = on ? 1 : 144; });
+var slowState = new SavedState("slow", false, (on) => { framesPerUpdate = on ? 60 : 0; });
+
+function handleClick(checkbox) {
+  var stateName = checkbox.dataset.state;
+  var state = window[stateName];
```

Comment:
I'm not a big fan of looking things up in "window"
can we have an explicit map from checkbox to state so we can do:

    var state = state_by_checkbox[checkbox];

instead?
We should be able to add the state to the state_by_checkbox dict from inside the SavedState constructor.